Sources/Caching/DeviceCache.swift

```diff
@@ -274,17 +274,34 @@ class DeviceCache {
 
     // MARK: - attribution
 
-    func latestNetworkAndAdvertisingIdsSent(appUserID: String) -> [String: String] {
+    func latestAdvertisingIdsByNetworkSent(appUserID: String) -> [AttributionNetwork: String] {
         return self.userDefaults.read {
             let key = CacheKeyBases.attributionDataDefaults + appUserID
-            let latestNetworkAndAdvertisingIdsSent = $0.object(forKey: key) as? [String: String] ?? [:]
-            return latestNetworkAndAdvertisingIdsSent
+            let latestAdvertisingIdsByRawNetworkSent = $0.object(forKey: key) as? [String: String] ?? [:]
+
+            // convert keys from UserDefault from Integer String to AttributionNetwork
+            let latestSent: [AttributionNetwork: String] =
+                latestAdvertisingIdsByRawNetworkSent.reduce(into: [:]) { adIdsByNetwork, adIdByRawNetworkString in
+                    if let networkRawValue = Int(adIdByRawNetworkString.key),
+                       let attributionNetwork = AttributionNetwork(rawValue: networkRawValue) {
+                    adIdsByNetwork[attributionNetwork] = adIdByRawNetworkString.value
+                } else {
+                    Logger.error(Strings.attribution.latest_attribution_sent_user_defaults_invalid)
+                }
+            }
```

Comment:
#1734 is in `main` now so this can be simplified:
```suggestion
           let latestSent: [AttributionNetwork: String] =
                latestAdvertisingIdsByRawNetworkSent.compactMapKeys { 
                    guard let networkRawValue = Int(adIdByRawNetworkString.key),
                       let attributionNetwork = AttributionNetwork(rawValue: networkRawValue) else {
                            Logger.error(Strings.attribution.latest_attribution_sent_user_defaults_invalid)    
                            return nil
                       }

                       return attributionNetwork
                   }
```